### ISSUE TITLE
No visible label for search input on small widths

### ISSUE THREAD

**Issue by BlueSlug** [inclusive-design/covid-data-monitor#24]
## Describe the bug
The Search input field for the map does not have a visible label at smaller viewport widths (i.e. primarily on mobile devices) and therefore lacks context. This also causes ambiguity with the site-wide search in the main navigation immediately above it.

### To reproduce
1. Go to https://wecount.inclusivedesign.ca/map on a mobile device or in a browser with a narrow viewport
2. Observe that the search input does not have a visible label

### Expected behavior
The search bar should have a visible label

## Screenshots
![Screenshot_20210609-001448_cropped](https://user-images.githubusercontent.com/547933/121973229-c9180280-cd4a-11eb-9810-0d85d9a6b90f.png)

### Desktop
 - OS: Windows 10 Pro v20H2
 - Browser: Chrome 91, Firefox 89
 - Affected version: 0.0.4

### Smartphone / tablet 
 - Device: Pixel 5
 - OS: Android 11
 - Browser: Chrome 91
 - Affected version: 0.0.4

**Comment by klown** [comment]
There are a couple of related a11y issues regarding this search input.  First, although there is a `<label>` for it, there is a mismatch between the <label>'s `@for` attribute, `for="search-query"`, and the search input's `@id`, `id="fl-search-query"`.  They almost match, except for the "fl-" prefix on the input's id.

Secondly, the magnifying glass shows up in the a11y tree as a graphic, but has no alternative text.  It might be okay to mark this `@role="presentation"` since it's not necessarily important for a screen reader user.  Otherwise, it should have alternative text.  I'm not sure what text to use here for the text.  Visual users see a magnifying glass which is a common cue that this a search input.  So, the text might be a generic "magnifying glass" or the more specific "search". Note that the latter is redundant given the label is "Search by postal code or city", and suggests the magnifying glass is somewhat decorative, i.e.,  role="presentation".

Whatever the text is, the markup to add it is: `<svg role="img" arial-label="magnifying glass" ...>`

Or, if it's just decorative: `<svg role="presentation" ...>`

**Comment by uttaraghodke** [comment]
![image](https://user-images.githubusercontent.com/66276957/122960316-4c3fe680-d351-11eb-8190-7e425468ef0a.png)

**Comment by uttaraghodke** [comment]
![image](https://user-images.githubusercontent.com/66276957/122960449-6974b500-d351-11eb-9a46-91e8b6512af5.png)

**Comment by uttaraghodke** [comment]
@cindyli We wont be able to have the floating 'search' bar anymore. Because we are lacing the label in mobile view as well

**Comment by cindyli** [comment]
> @cindyli We wont be able to have the floating 'search' bar anymore. Because we are lacing the label in mobile view as well

Thanks for the new design, @uttaraghodke. It looks great. I will work on it.

**Comment by klown** [comment]
@uttaraghodke Your screen shots look good to me.
